keras_nlp/layers/modeling/rotary_embedding.py

```diff
@@ -16,6 +16,40 @@
 
 
 class RotaryEmbedding(keras.layers.Layer):
+    """Rotary positional encoding layer.
+
+    This layer encodes absolute positional information with rotation
+    matrix. It calculates the rotary encoding with a mix of sine and
+    cosine functions with geometrically increasing wavelengths.
+    Defined and formulated in [RoFormer: Enhanced Transformer with Rotary Position Embedding](https://arxiv.org/abs/2104.09864v4).
+    Takes as input the query and key tensors. The input must have shape
```

Comment:
I think we should also allow this layer to take in shape `[batch_size, sequence_length, feature_dim]`. I'll leave some more comments below.

Reply:
sgtm.

Reply:
This still need updating in the docstring.